src/erlcloud_s3.erl

```diff
@@ -1156,3 +1165,35 @@ aws_region_from_host(Host) ->
         _ ->
             "us-east-1"
     end.
+
+%%
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/Redirects.html
+%% Note: redirect responce is handled only once.
+s3_follow_redirect({error, {http_error, _StatusCode, _StatusLine, ErrBody, ErrHeaders}} = Response, 
+    Config, Method, Bucket, Path, Subresource, Params, POSTData, Headers) ->
+    case Config#aws_config.s3_follow_redirect of
+        true ->
+            S3RegionEndpoint = case proplists:get_value("x-amz-bucket-region", ErrHeaders) of
+                undefined ->
+                    RedirectHostName = try 
+                        %% Try to get redirect location from error message.
+                        XML = element(1,xmerl_scan:string(binary_to_list(ErrBody))),
```

Comment:
OK. let's look for location header first and parse error message body at the end.